File: Esempi/03-Primes/js/primes.js
function sieveOfEratosthenes(n) {
  let primes = new Array(n + 1);
  for (let i = 2; i <= n; i++) {
    primes[i] = true;
  }

  for (let p = 2; p * p <= n; p++) {
    if (primes[p] === true) {
      for (let i = p * 2; i <= n; i += p) {
        primes[i] = false;
      }
    }
  }

  let result = 0;
  for (let i = 2; i <= n; i++) {
    if (primes[i]) {
      result++;
    }
  }

  return result;
}

let n = 100000000;
let result = sieveOfEratosthenes(n);
console.log(`Primes numbers less than ${n}: ${result}`);

File: Esempi/03-Primes/c/primes.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <stdbool.h>

#define N 120000000

int main()
{
  // printf("Prime numbers less than or equal to %d:\n", N);

  int64_t result = 0;

  bool *prime = (bool *)malloc((N + 1) * sizeof(bool));

  for (int i = 0; i < N; i++)
    prime[i] = true;

  for (int64_t p = 2; p * p <= N; p++)
  {
    // If prime[p] is not changed, then it is a prime
    if (prime[p] == true)
    {
      // Update all multiples of p
      for (int i = p * 2; i <= N; i += p)
        prime[i] = false;
    }
  }

  for (int64_t p = 2; p <= N; p++)
    if (prime[p])
      result++;

  // printf("%ld\n", result);
  return 0;
}